ProcessLib/BoundaryCondition/Python/PythonBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,198 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include "PythonBoundaryCondition.h"
+
+#include "MathLib/LinAlg/Eigen/EigenMapTools.h"
+#include "NumLib/DOF/DOFTableUtil.h"
+#include "ProcessLib/BoundaryCondition/GenericNaturalBoundaryConditionLocalAssembler.h"
+
+#include "PythonBoundaryConditionPythonSideInterface.h"
+
+namespace ProcessLib
+{
+//! Can be thrown to indicate that a member function is not overridden in a
+//! derived class (in particular, if a Python class inherits from a C++ class).
+struct MethodNotOverriddenInDerivedClassException
+{
+};
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+class PythonBoundaryConditionLocalAssembler final
+    : public GenericNaturalBoundaryConditionLocalAssembler<
+          ShapeFunction, IntegrationMethod, GlobalDim>
+{
+    using Base = GenericNaturalBoundaryConditionLocalAssembler<
+        ShapeFunction, IntegrationMethod, GlobalDim>;
+
+public:
+    PythonBoundaryConditionLocalAssembler(
+        MeshLib::Element const& e,
+        std::size_t const /*local_matrix_size*/,
+        bool is_axially_symmetric,
+        unsigned const integration_order,
+        PythonBoundaryConditionData const& data)
+        : Base(e, is_axially_symmetric, integration_order),
+          _data(data),
+          _element(e)
+    {
+    }
+
+    void assemble(std::size_t const boundary_element_id,
+                  NumLib::LocalToGlobalIndexMap const& dof_table_boundary,
+                  double const t, const GlobalVector& x, GlobalMatrix& /*K*/,
+                  GlobalVector& b, GlobalMatrix* Jac) override
+    {
+        using ShapeMatricesType =
+            ShapeMatrixPolicyType<ShapeFunction, GlobalDim>;
+        using FemType =
+            NumLib::TemplateIsoparametric<ShapeFunction, ShapeMatricesType>;
+        FemType fe(*static_cast<const typename ShapeFunction::MeshElement*>(
+            &_element));
+
+        unsigned const num_integration_points =
+            Base::_integration_method.getNumberOfPoints();
+        auto const num_var = _data.dof_table_bulk.getNumberOfVariables();
+        auto const num_nodes = _element.getNumberOfNodes();
+        auto const num_comp_total =
+            _data.dof_table_bulk.getNumberOfComponents();
+
+        auto const& bulk_node_ids_map =
+            *_data.boundary_mesh.getProperties().getPropertyVector<std::size_t>(
+                "bulk_node_ids");
+
+        // gather primary variables
+        Eigen::MatrixXd primary_variables_mat(num_nodes, num_comp_total);
```

Comment:
Optionally use compile time size for nodes' size...

Reply:
Hm... I think, I'd like to leave it all dynamic:
`Eigen::Matrix<double, ShapeFunction::size /*?*/, Eigen::Dynamic>` doesn't really look more pleasant.